### PR TITLE
String setters will throw IllegalArgumentException

### PR DESCRIPTION
**Pull request by kneth**
As suggested in #1756, `IllegalArgumentException` is better than `RealmError` when trying to store an invalid string.

@realm/java @remeissner

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 1]
```diff
@@ -119,6 +119,35 @@ public void stringEncoding() {
...
+        realm.clear(AllTypes.class);
+        realm.commitTransaction();
+
+        realm.beginTransaction();
+        try {
+            AllTypes highSurrogate = realm.createObject(AllTypes.class);
+            highSurrogate.setColumnString(high);
+            fail();
+        }
+        catch (IllegalArgumentException ignored) {}
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: C++ catch style, move one up

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 2]
```diff
@@ -119,6 +119,35 @@ public void stringEncoding() {
...
+        realm.cancelTransaction();
+
+        realm.beginTransaction();
+        try {
+            AllTypes lowSurrogate = realm.createObject(AllTypes.class);
+            lowSurrogate.setColumnString(low);
+            fail();
+        }
+        catch (IllegalArgumentException ignored) {}
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: Same

**Comment by cmelchior** [comment]
Except from small comments :+1:

**Comment by dalinaum** [comment]
Beside @cmelchior 's comment, :+1:

**Comment by beeender** [comment]
:+1:

**Pull request merged**